docs/docs/algorithms/coloring/greedy-graph-coloring.md

```diff
@@ -25,6 +25,34 @@ If no coloring is possible, an empty `unordered_map` is returned. This is the ca
 ```cpp
 template <typename GRAPH>
 std::unordered_map<vertex_id_t, int> greedy_graph_coloring(const GRAPH& graph);
+    // Step 1: Sort vertices by degree in descending order
+    std::vector<DegreeVertexPair> degree_vertex_pairs;
+    for (const auto& vertex : graph.get_vertices()) {
+        int degree = graph.get_degree(vertex);
+        degree_vertex_pairs.emplace_back(degree, vertex);
+    }
+
+    std::sort(degree_vertex_pairs.rbegin(), degree_vertex_pairs.rend());
+
+    for (const auto& degree_vertex_pair : degree_vertex_pairs) {
+        vertex_id_t current_vertex = degree_vertex_pair.second;
+        int color = 1;  // Start with color 1
+
+        // Check colors of adjacent vertices
+        for (const auto& neighbor : graph.get_adjacent_vertices(current_vertex)) {
+            if (color_map.find(neighbor) != color_map.end()) {
+                // If neighbor is already colored with this color, increment the color
+                if (color_map[neighbor] == color) {
+                    color++;
+                }
+            }
+        }
+
+        // Assign the color to the current vertex
+        color_map[current_vertex] = color;
+    }
+
+    return color_map;
```

Comment:
This is the markdown file, meaning that we only have the syntax of the function call and a description here.
Your code itself needs to live in new files `include/graaflib/algorithm/coloring/welsh_powell.h` and `include/graaflib/algorithm/coloring/welsh_powell.tpp` similar to the other algorithms :slightly_smiling_face: